airbyte-workers/src/main/java/io/airbyte/workers/DefaultSyncWorker.java

```diff
@@ -82,6 +88,22 @@ public DefaultSyncWorker(final Source<T> source,
 
       destination.notifyEndOfStream();
 
+      final boolean shouldNormalize = Optional
+          .ofNullable(syncInput.getDestinationConnection().getConfiguration().get(WorkerConstants.BASIC_NORMALIZATION_KEY))
+          .map(JsonNode::asBoolean)
+          .orElse(false);
+      if (shouldNormalize) {
+        LOGGER.info("Running normalization.");
+        final Path normalizationRoot = Files.createDirectories(jobRoot.resolve("normalize"));
+        try {
+          if (!normalizationRunner.normalize(normalizationRoot, syncInput.getDestinationConnection().getConfiguration(), syncInput.getCatalog())) {
+            throw new WorkerException("Normalization Failed.");
+          }
+        } finally {
+          normalizationRunner.close();
```

Comment:
this feels ham fisted. ideally we would treat the normalization runner as a `try-resource`, but because it is instantiated outside of this class we can't really do this. it makes the contract nebulous though, because it means we only try to close it if we at least attempt to call `normalize`.

other options:
* Add a `start` method - this at least makes the contract clearer.
* Create a factory method for this and only call the factory once we're in here.

for the amount of code it would take to do either of these, i'm not really sure it is worth it, but i'd like a second opinion.

Reply:
having the start method makes the lifecycle of the object a bit simple and looks like the source/destination object that are already present.